readme update (#123)
diskin-lab-chop/AutoGVP: README.md

1. Run the [Kids First Germline Annotation Workflow](https://github.com/kids-first/kf-germline-workflow/blob/v0.4.4/docs/GERMLINE_SNV_ANNOT_README.md) first.
This workflow currently annotates variants with ClinVar (2022-05-07).
2. Run the [Pathogenicity Preprocessing Workflow](https://github.com/d3b-center/D3b-Pathogenicity-Preprocessing), which performs ANNOVAR with InterVar and AutoPVS1 annotations.

> 3. Download required db files

Reviewer: I think this can use more explanation and specificity. Eg: Download ClinVar variant and submission files, which will be used to .....

> 4. Run `select-clinVar-submissions.R` within `AutoGVP/`

Reviewer: Please add some additional information here as well to give a brief brief summary of what this script is doing.

python InterVar.py -b hg38 -i input/test_VEP.vcf --input_type=VCF -o test_VEP
```

> 4. Run [AutoPVS1 v2.0](https://github.com/JiguangPeng/autopvs1/releases/tag/v2.0).

Reviewer: Can one of you @naqvia or @rjcorb create a new issue that this will need to be changed since AutoPVS1 is not working out of the box. We might either have to supply a copy here or point to our copy in the KF repo.

5. Optional: provide a ClinVar VCF file. If not supplied by the user, the most recent ClinVar file will be downloaded with `download_db_files.sh` and used in AutoGVP.
6. Download required db files

> 7. Run `select-clinVar-submissions.R` within `AutoGVP/`

Reviewer: Once the same steps have additional description above, please update these.

> 7. Run `select-clinVar-submissions.R` within `AutoGVP/`

Reviewer 2: should we also note that if this has been run before and same ClinVar data is used, this doesn't need to run every time?

8. Run AutoGVP wrapper script using `--workflow="custom"`

AutoGVP Requirements (recommended to place all in the `input/` folder):
- VEP-annotated VCF (`*VEP.vcf`)
- ANNOVAR file (`*hg38_multianno.txt`)
- InterVar file (`*hg38_multianno.txt.intervar`)
- AutoPVS1 file (`*autopvs1.txt`)
- ClinVar VCF (`clinvar_yyyymmdd.vcf.gz` optional user input or `clinvar.vcf.gz` will be downloaded with `download_db_files.sh`)
- Variant submissions file (`ClinVar-selected-submissions.tsv` generated by `select-clinVar-submissions.R`)

Example run:

Download ClinVar files: 

```bash
bash download_db_files.sh
```

Run `select-clinVar-submissions.R`:

```
Rscript `select-clinVar-submissions.R` --variant_summary <variant_summary.txt.gz> --submission_summary <submission_summary.txt.gz>
```

Run AutoGVP:
```r
bash run_autogvp.sh --workflow="custom" \
--vcf=input/test_VEP.vcf \
--filter_criteria=<filter criteria>
--clinvar=input/clinvar.vcf.gz \
--intervar=input/test_VEP.hg38_multianno.txt.intervar \
--multianno=input/test_VEP.vcf.hg38_multianno.txt \
--autopvs1=input/test_autopvs1.txt \
--outdir=../results \
--out="test_custom"

> ```

Author: We do, but this is now hardcoded into the sh script. 

```
## default files
variant_summary_file="input/ClinVar-selected-submissions.tsv"
```

### Step by step workflow

* __Filter VCF file__. By default, AutoGVP filters based on `FILTER` column (`PASS` or `.`). Other criteria can be specified by `filter_criteria` argument as follows: 

```
filter_criteria="INFO/AF>=0.2 INFO/DP>=15"
```

* __Run Pathogenicity Assessment__. The R scripts `01-annotate_variants_CAVATICA_input.R` and `01-annotate_variants_custom_input.R` perform the following steps:

  1. Read in ClinVar-annotated VCF file
  2. Assign ClinVar stars based on `CLNREVSTAT`*
  3. For ClinVar variants, report `CLINSIG` as final call; resolve ambiguous variants (`criteria_provided,_conflicting_interpretations`) by checking against ClinVar variant submission file
  4. Identify variants that need further Intervar annotation and possible re-adjustment (variants with 0 stars or not in ClinVar database)

>   5. Load and merge multianno, interVar, and autopvs1 files

Reviewer: ```suggestion
  5. Load and merge ANNOVAR multianno, InterVar, and AutoPVS1 files
```

  6. Create columns for `evidencePVS1`, `evidencePS`, `evidencePM`, `evidencePP`, `evidenceBP`, `evidencePM` and `evidenceBA1` (variables that may need re-adjusting) by parsing `InterVar: InterVar and Evidence` column

>   7. Adjust evidence columns based on autopvs1 `criterion` column

Reviewer: ```suggestion
  7. Adjust evidence columns based on AutoPVS1 `criterion` column
```

>   8. Report Intervar final call (if unadjusted) or final call based on recalculated evidence variables (if adjusted)

Reviewer: ```suggestion
  8. Report InterVar final call (if unadjusted) or final call based on re-calculated evidence variables (if adjusted)
```

  9. Save output 

> * __Parse VCF file__. `parse_vcf.sh` converts the vcf file to a tsv file with tab-separated INFO fields column. 

Reviewer: ```suggestion
* __Parse VCF file__. `parse_vcf.sh` converts the VCF file to a TSV file with INFO fields as tab-separated columns. 
```

* __Resolve gene annotations and produce final output files__ (`04-filter_gene_annoations.R`)

>   1. Read in parsed vcf file, and select single VEP annotation based on `PICK` column (`PICK == 1`). See R script for criteria used to select pick transcripts. 

Reviewer: ```suggestion
  1. Read in parsed VCF file, and select single VEP annotation based on `PICK` column (`PICK == 1`). 
  See R script for criteria used to select pick transcripts. 
```

-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
chr1 | 1332490 | C | T | rs201607183 | TAS1R3 | missense_variant | chr1:g.1332490C>T | c.959C>T | p.Thr320Met | Uncertain_significance | ClinVar | 1 | Uncertain_significance | InterVar: Uncertain significance PVS1=0 PS=[0, 0, 0, 0, 0] PM=[1, 0, 0, 0, 0, 0, 0] PP=[0, 0, 1, 0, 0, 0] BA1=0 BS=[0, 0, 0, 0, 0] BP=[0, 0, 0, 0, 0, 0, 0, 0]
chr1 | 1390349 | C | T | rs769726291 | CCNL2 | missense_variant | chr1:g.1390349C>T | c.887G>A | p.Gly296Asp | Uncertain_significance | InterVar | NA | NA | InterVar: Uncertain significance PVS1=0 PS=[0, 0, 0, 0, 0] PM=[1, 1, 0, 0, 0, 0, 0] PP=[0, 0, 0, 0, 0, 0] BA1=0 BS=[0, 0, 0, 0, 0] BP=[0, 0, 0, 0, 0, 0, 0, 0]

*NOTE: gnomAD v.3.1.1 non-cancer AF popmax values (`gnomad_3_1_1_AF_non_cancer`) will also be included in abridged output when provided. 

See [here](https://github.com/diskin-lab-chop/AutoGVP/blob/main/AutoGVP/input/output_colnames.tsv) for list of columns included in full output

> ## Code Author

Reviewer: ```suggestion
## Code Authors
```
